lib/clientv3/ippool.go

```diff
@@ -63,9 +63,49 @@ func (r ipPools) Create(ctx context.Context, res *apiv3.IPPool, opts options.Set
 		return nil, err
 	}
 
+	// Check that there are no existing blocks in the pool range that have a different block size.
+	poolBlockSize := res.Spec.BlockSize
+	poolIP, poolCIDR, err := net.ParseCIDR(res.Spec.CIDR)
+	if err != nil {
+		return nil, cerrors.ErrorParsingDatastoreEntry{
+			RawKey:   "CIDR",
+			RawValue: string(res.Spec.CIDR),
+			Err:      err,
+		}
+	}
+
+	ipVersion := 4;
+	if poolIP.To4() == nil {
+		ipVersion = 6
+	}
+
+	blocks, err := r.client.backend.List(ctx, model.BlockListOptions{IPVersion: ipVersion}, "")
+	// If the error is OperationUnsupported then skip the blockSize check - IPAM is not supported on KDD.
+	if _, ok := err.(cerrors.ErrorOperationNotSupported); !ok && err != nil {
```

Comment:
Does this work when err is an OperationNotSupported? Won't we go into the next branch and try to access a nil pointer?

Reply:
Yup I think that looks good now.